### DISCUSSION TITLE
Unit of Atomic or Global Polarizability

### DISCUSSION THREAD

**Discussion by hanao2** [deepmodeling/deepmd-kit#2906] [General]
Hello,

I am going through the example files for modeling atomic/global polarizabilities. My question is what the unit of polarizability data should be? I read through the reference paper (https://pubs.rsc.org/en/content/articlelanding/2020/cp/d0cp01893g) and noticed that their unit is e.A/a.u.; is this the same unit as the one used in the example files? Thank you!

**Comment by wanghan-iapcm** [comment]
Hello, the unit does not matter. You may use the parameter [`scale`](https://docs.deepmodeling.com/projects/deepmd/en/master/train-input-auto.html#model-fitting-net-polar) to scale the output of fitting net to any unit you want. The scale is recommend to be chosen so that the output of the fitting net has standard deviation of 1.0.

**Reply by hanao2** [reply]
Hello,
Thanks for your response. I have another question about the 'fit_diag' key of the 'polar' model. When setting 'fit_diag=False' do we mean that only the non-diagonal elements of the polarizability matrix are being fitted? Also, with 'fit_diag=True' do we mean that only the diagonal elements are being fitted? 
Thanks!

**Reply by wanghan-iapcm** [reply]
please ref to [this paper ](https://arxiv.org/abs/2004.07369).  
![image](https://github.com/deepmodeling/deepmd-kit/assets/92130845/fafe5b1f-a30a-45fc-87eb-755d17dc6e29)
`fit_diag` means only the diagonal part of matrix F, i.e. F_jj is used to form the polarizability, as shown by the equation in the screen shot. 
Otherwise, the full matrix F is used and one will have two contraction indices, one with the left T and the other with the right T. 
The `fit_diag=True` option would be accurate enough according to our experience with water polarizability.

**Reply by hanao2** [reply]
Hello,

Thanks for your response. It is very clear now. In the mentioned paper, I notice the use of wannier centroid (wc) displacement to calculate the polarizability tensor. May I ask if this (use of wc-based approach) is necessary for the DeepPolar model to be applicable? or one could use other methods of quantifying the electronic structure perturbation as well? 

Thanks!
